driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/OperationExecutorImpl.java

```diff
@@ -63,40 +68,32 @@
         notNull("readPreference", readPreference);
         notNull("callback", callback);
         final SingleResultCallback<T> errHandlingCallback = errorHandlingCallback(callback, LOGGER);
-        clientSessionHelper.withClientSession(session, this, new SingleResultCallback<AsyncClientSession>(){
-            @Override
-            public void onResult(final AsyncClientSession clientSession, final Throwable t) {
-                if (t != null) {
-                    errHandlingCallback.onResult(null, t);
-                } else {
-                    getReadWriteBinding(readPreference, readConcern, clientSession,
-                            session == null && clientSession != null,
-                            new SingleResultCallback<AsyncReadWriteBinding>() {
-                                @Override
-                                public void onResult(final AsyncReadWriteBinding binding, final Throwable t) {
-                                    if (t != null) {
-                                        errHandlingCallback.onResult(null, t);
-                                    } else {
-                                        if (session != null && session.hasActiveTransaction()
-                                                && !binding.getReadPreference().equals(primary())) {
-                                            binding.release();
-                                            errHandlingCallback.onResult(null,
-                                                    new MongoClientException("Read preference in a transaction must be primary"));
+        clientSessionHelper.withClientSession(session, this, (clientSession, t) -> {
```

Comment:
I was expecting to see this un-callbacked.  Is it possible?

Reply:
Will try to once the OperationExecutor has been lifted it should be do able.

Reply:
Done